tests/array_test.go

```diff
@@ -66,6 +66,50 @@ func TestSimpleArray(t *testing.T) {
 	require.NoError(t, rows.Err())
 }
 
+type customArr []customStr
+
+func TestCustomArray(t *testing.T) {
+	conn, err := GetNativeConnection(nil, nil, &clickhouse.Compression{
+		Method: clickhouse.CompressionLZ4,
+	})
+	ctx := context.Background()
+	require.NoError(t, err)
+	const ddl = `
+		CREATE TABLE test_array (
+			  Col1 Array(Enum ('hello'   = 1,  'world' = 2)),
+			  Col2 Array(String)
+		) Engine MergeTree() ORDER BY tuple()
```

Comment:
Updated test: added Array(Enum) and Array(String)
to show simple []customStrArray not only enum